hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/test/TestSources.java

```diff
@@ -161,4 +187,59 @@ void fillBuffer(TimestampedSourceBuffer<T> buf) throws Exception {
             }
         }
     }
+
+    /**
+     * Returns a {@link com.hazelcast.jet.pipeline.StreamSource} for {@code
+     * long} values. The value creation is distributed across the cluster and
+     * is done using {@link StreamSourceLong}. All {@link StreamSourceLong}
+     * instances are created identically, one per cluster member. The returned
+     * source is designed to be used for high-throughput performance testing.
+     *
+     * @param itemsPerSecond how many items should be emitted each second
+     * @param initialDelay initial delay before emitting values
+     *
+     * @return {@link com.hazelcast.jet.pipeline.StreamSource} with {@code long} values
+     *          created in a distributed fashion
+     *
+     * @since 4.3
+     *
+     */
+    @Nonnull
+    public static StreamSource<Long> streamSourceLong(long itemsPerSecond, long initialDelay) {
+        return streamSourceLong(itemsPerSecond, initialDelay, Vertex.LOCAL_PARALLELISM_USE_DEFAULT);
+    }
+
+    /**
+     * Returns a {@link com.hazelcast.jet.pipeline.StreamSource} for {@code
+     * long} values. The value creation is distributed across the cluster and
+     * is done using {@link StreamSourceLong}. All {@link StreamSourceLong}
+     * instances are created identically, one per cluster member. The returned
+     * source is designed to be used for high-throughput performance testing.
+     *
+     * @param itemsPerSecond how many items should be emitted each second
+     * @param initialDelay initial delay before emitting values
+     * @param preferredLocalParallelism the preferred local parallelism
+     *
+     * @return {@link com.hazelcast.jet.pipeline.StreamSource} with {@code long} values
+     *          created in a distributed fashion
+     *
+     * @since 4.3
+     *
+     */
+    @Nonnull
+    public static StreamSource<Long> streamSourceLong(
+            long itemsPerSecond, long initialDelay, int preferredLocalParallelism
```

Comment:
`preferredLocalParallelism`  can be already set at source stage level. What does it bring here?

Reply:
As far as I can see, you can set the _localParallelism_ at the stage level. Both, _localParallelism_ and _preferredLocalParallelism_ are considered in _Vertex.determineLocalParallelism(int defaultParallelism)_:
_/**
     * Determines the local parallelism value for the vertex by looking
     * its local parallelism and meta supplier's preferred local parallelism.
     * If none of them is set, returns the provided default parallelism
     */_

Does that answer your question?

Reply:
I guess we could remove `preferredLocalParallelism` from here, we have this feature for custom source implementations that can declare their preferred parallelism. But, since this is for ephemeral sources, we don't need it and can use `sourceStage.setLocalParallelism()`

Reply:
removed with d476f97f6c1cb3bdd945b2b0da124869086022da